### ISSUE TITLE
Navigating to a specific deployment shows a blank page and a console error

### ISSUE THREAD

**Issue by filpano** [PrefectHQ/prefect#8801]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I refreshed the page and this issue still occurred.
- [X] I checked if this issue was specific to the browser I was using by testing with a different browser.

### Bug summary

I created a Deployment using the prefect CLI as follows (newlines for readability):

```
prefect deployment build --name "masterdata_export" \
--description "..." \
--version "1" \
--tag "masterdata" \
--work-queue "default" \
--storage-block "s3/deployment-definitions" \
--infra-block "docker-container/infrastructure-masterdata-export" \
--timezone "Europe/Berlin" \
--apply \
extract_data.py:read_and_save_data
```
When clicking on a specific deployment from the _Deployments_ screen, I am navigated to a blank page and a console error occurs. After this, I have to navigate away _twice_ (e.g. `Flow -> Flow runs`, `Blocks -> Deployments`) for the UI to render correctly again.

Interestingly, there seems to be another, related issue when navigating from within the Flow and then navigating to the Deployment tab and clicking the deployment, which leads to another blank page and a different console error which now includes an HTTP `422 Unprocessable Entity` error.

From a cursory glance at the error itself, it seems that the deployment `id` is not being properly propagated to the actual request.

See screen shots below.

### Reproduction

Create a deployment, then either:

1. Navigate to the _Deployments_ page. 
2. Click on the deployment.
3. A blank page is displayed and a console error occurs.

or:

1. Navigate to the _Flows_ page.
2. Click on the _Deployment_ tab.
3. Click on the deployment.
4. A blank page is displayed and a console error occurs.

### Error

When clicking on a specific deployment from the _Deployments_ screen,
![image](https://user-images.githubusercontent.com/5376771/225273917-2e2a19c3-113a-4523-928a-9658d3cab6ce.png)

I am navigated to a blank page:
![image](https://user-images.githubusercontent.com/5376771/225273977-b7892e27-80d2-4b45-b268-2585625567aa.png)

after this, I have to navigate away _twice_ (e.g. `Flow -> Flow runs`, `Blocks -> Deployments`) for the UI to render again.

I see the following error message in the console:
![image](https://user-images.githubusercontent.com/5376771/225274453-848b25ef-b12e-4b8f-b567-d80888959eaf.png)

---

Interestingly, there seems to be another, related issue when navigating from within the Flow.

Going to the Deployment flow:

![image](https://user-images.githubusercontent.com/5376771/225275746-72164954-d792-47eb-97e9-243b455f39ae.png)

and then navigating to Deployments and clicking the deployment:

![image](https://user-images.githubusercontent.com/5376771/225275908-10eea577-b471-467e-ba85-f88885bd119b.png)

leads to another blank page and the following, different console error:
![Screenshot from 2023-03-15 11-04-35](https://user-images.githubusercontent.com/5376771/225276743-0c4d450f-609c-493b-9413-3363aa2cd2c7.png)

![Screenshot from 2023-03-15 12-08-35](https://user-images.githubusercontent.com/5376771/225291877-1e83028a-5ae1-4433-8bb8-f51394f0c9f4.png)

including an HTTP `422 Unprocessable Entity` error this time around.

The actual error response of the requests is as follows:

`count`
![image](https://user-images.githubusercontent.com/5376771/225292215-f4597801-1a63-4278-8b04-6dfee85853c1.png)

![image](https://user-images.githubusercontent.com/5376771/225277424-e41b28af-e980-4c33-a2f4-1bd3c2776689.png)

`filter`
![image](https://user-images.githubusercontent.com/5376771/225292270-05ce263e-47e5-4a37-91ec-e972fff0bf3c.png)

![image](https://user-images.githubusercontent.com/5376771/225277545-bf3ae180-8722-4044-8c11-d89351e7ee0a.png)

### Browsers

- [X] Chrome
- [X] Firefox
- [ ] Safari
- [ ] Edge

### Prefect version

```Text
Version:             2.8.4
API version:         0.8.4
Python version:      3.10.6
Git commit:          f09ccc32
Built:               Thu, Mar 2, 2023 12:34 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         server
```
Browser versions tested:

```
Chrome Version 107.0.5304.110 (Official Build) (64-bit)
Firefox 111.0 (64-bit)
```

### Additional context

Bug did not previously occur on prefect `2.7.7`.

**Comment by zhen0** [comment]
Hi @filpano - I believe this issue was closed by https://github.com/PrefectHQ/prefect/issues/8712 and should be fixed if you update to 2.8.5. Please let us know in a new issue if you continue to see it after updating.

**Comment by filpano** [comment]
@zhen0 Updating to 2.8.5 fixes the problem. Thanks for the quick response! My apologies, It seems I only searched for open issues and not closed ones. :)
